lark/exceptions.py

```diff
@@ -109,32 +110,54 @@ def match_examples(self, parse_fn, examples, token_type_match_fallback=False, us
                             candidate = label, False
 
         return candidate[0]
+    
+    def _format_terminals(self, names):
+        if self._all_terminals:
+            t = []
+            for name in names:
+                try:
+                    t.append(next(t.nice_print for t in self._all_terminals if t.name == name))
```

Comment:
Looks like `all_terminals` should be a dict.

Reply:
Yeah, but since this should only be executed when we actually print the error message, I decided that it wasn't worth the hassle (But maybe I overlook something and we just have dictionaries lying around anyway. The inefficiency of this code is way I decided to make it lazy.

Reply:
This isn't about performance, it's about readability.